source/particle/world.cc

```diff
@@ -0,0 +1,889 @@
+/*
+ Copyright (C) 2015 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#include <aspect/particle/world.h>
+#include <aspect/global.h>
+#include <aspect/utilities.h>
+#include <aspect/compat.h>
+#include <aspect/geometry_model/box.h>
+
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/grid/grid_tools.h>
+#include <boost/serialization/map.hpp>
+#include <boost/archive/text_oarchive.hpp>
+#include <boost/archive/text_iarchive.hpp>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    template <>
+    World<2>::World()
+    {
+      data_offset = numbers::invalid_unsigned_int;
+      integrator = NULL;
```

Comment:
if these are member variables, initialize them in the member initializer list (after a colon)